plugins/inputs/t128_graphql/response_processor_test.go

```diff
@@ -0,0 +1,252 @@
+		ExpectedError: nil,
+	},
+	{
+		Name:   "converts tag to string if numeric",
+		Fields: map[string]string{"test-field": "test-field"},
+		Tags:   map[string]string{"test-tag": "test-tag"},
+		JsonInput: []*gabs.Container{
+			generateJsonTestData([]byte(`{
+				"test-field": 128,
+				"test-tag": 128
+			}`)),
+		},
+		ExpectedOutput: []*plugin.ProcessedResponse{
+			&plugin.ProcessedResponse{
+				Fields: map[string]interface{}{"test-field": 128.0},
+				Tags:   map[string]string{"test-tag": "128"},
+			},
+		},
+		ExpectedError: nil,
+	},
+	{
+		Name:   "renames tags and fields",
+		Fields: map[string]string{"test-field-renamed": "test-field"},
+		Tags:   map[string]string{"test-tag-renamed": "test-tag"},
+		JsonInput: []*gabs.Container{
+			generateJsonTestData([]byte(`{
+				"test-field": 128,
+				"test-tag": 128
+			}`)),
+		},
+		ExpectedOutput: []*plugin.ProcessedResponse{
+			&plugin.ProcessedResponse{
+				Fields: map[string]interface{}{"test-field-renamed": 128.0},
+				Tags:   map[string]string{"test-tag-renamed": "128"},
+			},
+		},
+		ExpectedError: nil,
+	},
+	{
+		Name:   "process response with multiple nodes",
+		Fields: map[string]string{"test-field": "test-field"},
+		Tags:   map[string]string{"test-tag": "test-tag"},
+		JsonInput: []*gabs.Container{
+			generateJsonTestData([]byte(`{
+				"test-field": 128,
+				"test-tag": "test-string-1"
+			}`)),
+			generateJsonTestData([]byte(`{
+				"test-field": 95,
+				"test-tag": "test-string-2"
+			}`)),
+		},
+		ExpectedOutput: []*plugin.ProcessedResponse{
+			&plugin.ProcessedResponse{
+				Fields: map[string]interface{}{"test-field": 128.0},
+				Tags:   map[string]string{"test-tag": "test-string-1"},
+			},
+			&plugin.ProcessedResponse{
+				Fields: map[string]interface{}{"test-field": 95.0},
+				Tags:   map[string]string{"test-tag": "test-string-2"},
+			},
+		},
+		ExpectedError: nil,
+	},
+	{
+		Name:   "process response with nested tags",
+		Fields: map[string]string{"test-field": "test-field"},
+		Tags:   map[string]string{"test-tag-1": "test-tag-1", "test-tag-2": "state/test-tag-2"},
+		JsonInput: []*gabs.Container{
+			generateJsonTestData([]byte(`{
+				"test-field": 128,
+				"test-tag-1": "test-string-1",
+			  	"state": {
+				  "test-tag-2": "test-string-2"
+			  	}
+		  	}`)),
+			generateJsonTestData([]byte(`{
+				"test-field": 95,
+				"test-tag-1": "test-string-3",
+				"state": {
+					"test-tag-2": "test-string-4"
+				}
+			}`)),
+		},
+		ExpectedOutput: []*plugin.ProcessedResponse{
+			&plugin.ProcessedResponse{
+				Fields: map[string]interface{}{"test-field": 128.0},
+				Tags:   map[string]string{"test-tag-1": "test-string-1", "test-tag-2": "test-string-2"},
+			},
+			&plugin.ProcessedResponse{
+				Fields: map[string]interface{}{"test-field": 95.0},
+				Tags:   map[string]string{"test-tag-1": "test-string-3", "test-tag-2": "test-string-4"},
+			},
+		},
+		ExpectedError: nil,
+	},
+	{
+		Name:   "process response with multi-level nested tags",
```

Comment:
I didn;t spot a test case for verifying the nested handling for fields? You should also add one where the nesting exists at both levels with some overlap, so something like this

```
		Fields: map[string]string{"test-field": "state1/state2/field-1"},
		Tags:   map[string]string{"test-tag-1": "test-tag-1", "test-tag-2": "state1/state2/state3/test-tag-2"},
```